src/net_processing.cpp

```diff
@@ -798,6 +803,11 @@ void PeerLogicValidation::BlockChecked(const CBlock& block, const CValidationSta
                 Misbehaving(it->second.first, nDoS);
         }
     }
+    else if (state.IsValid() && !IsInitialBlockDownload()) {
+        if (it != mapBlockSource.end() && State(it->second.first)) {
+            MaybeSetPeerAsAnnouncingHeaderAndIDs(it->second.first, *connman);
```

Comment:
Ideally this would go in UpdatedBlockTip instead. mapBlockSource wouldn't be available there, but BlockChecked isnt really capturing where we want this - we want to know that this is actually the best currently available block, not just a block that was just checked.

Not really sure how to fix this - maybe just a mapBlocksInFlight.empty() is sufficient, but its still super ugly.